internal/codegen/codegen.go

```diff
@@ -96,7 +100,7 @@ func generatePrincipalPolicy(parent *policyv1.Policy, p *policyv1.PrincipalPolic
 
 	for i, rule := range p.Rules {
 		if err := rg.AddPrincipalRule(rule); err != nil {
-			return nil, newRuleGenErr(parent, i+1, err)
+			return nil, newRuleGenErr(parent, i+1, rule.Resource, err)
```

Comment:
nit:
A principal rule resource is used as a rule name, so an error message might be:
`Failed to generate code for rule 'leave_request' (#3)`. 

What do you think about having a separate `newPrincipalRuleGenErr` function producing the following?
`Failed to generate code for rule (#3) associated with resource 'leave_request'`.

Reply:
Good idea.